### PR TITLE
[DOCS] Updating password usage for create user API

### LINKED ISSUE #62759: Create or update users API docs incorrectly state that you cannot change user password
**Elasticsearch version** (`bin/elasticsearch --version`): 7.9.0

**Plugins installed**: [none]

**JVM version** (`java -version`): bundled with Ubuntu 20.04 ES package

**OS version** (`uname -a` if on a Unix-like system): Ubuntu 20.04

**Description of the problem including expected versus actual behavior**:
https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-user.html
Actual behavior:
"Create or update users API" can update password which is not allowed according to documentation.
Expected behavior:
Password will not be changed.

**Steps to reproduce**:

 1. Create user using POST or PUT API reuquest according to  https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-user.html
 2. Use same query asi in 1) and also supply password field with different password than in step 1)
 3. Password of the user has been changed to password from step 2)

**Provide logs (if relevant)**:

### PR DESCRIPTION
**Pull request by lockewritesdocs**
This PR clarifies the password usage guidance for the [create users API](https://elasticsearch_63292.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-api-put-user.html#security-api-put-user-desc), and also marks the `password` as required.

Closes #62759 

Backports:
* 7.10 #63440
* 7.x #63441
* 7.9 #63442

### CODE REVIEW

File: x-pack/docs/en/rest-api/security/create-users.asciidoc [hunk 1]
```diff
@@ -73,7 +73,7 @@ The following parameters can be specified in the body of a POST or PUT request:
 (object) Arbitrary metadata that you want to associate with the user.
 
 `password`::
-(string) The user's password. Passwords must be at least 6 characters long.
+(Required, string) The user's password. Passwords must be at least 6 characters long.
```
Comment (lcawl) [inline x-pack/docs/en/rest-api/security/create-users.asciidoc:76]: In some other APIs where a parameter is required in some but not all situations, we've added an asterisk like this:
```suggestion
(Required*, string) The user's password. Passwords must be at least 6 characters long.
```
E.g. https://www.elastic.co/guide/en/elasticsearch/reference/7.9/analysis-dict-decomp-tokenfilter.html

File: x-pack/docs/en/rest-api/security/create-users.asciidoc [hunk 2]
```diff
@@ -26,12 +26,12 @@ to as _native users_.
 [[security-api-put-user-desc]]
 ==== {api-description-title}
 
-When updating a user, you can update everything but its `username` and `password`. 
-To change a user's password, use the 
-<<security-api-change-password, change password API>>.
+A `password` is required for adding a new user but is optional when updating an
+existing user. To change a user's password,
```
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/create-users.asciidoc:30]: This reads a little strangely to me now - it says _you can optionally change the password here_ but also _you should use this other API_.
 
```suggestion
existing user. To change a user's password without updating any other fields,
```

**Review by tvernum — APPROVED** [review]

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Comment by lockewritesdocs** [comment]
@elasticmachine test this please

**Pull request merged**